fix(stirling-pdf): service failing with H2 database schema error

## Issue Summary

Stirling PDF service is failing to start due to a H2 database schema mismatch. The service has been failing since Aug 27 04:01:28 BST.

## Error Details

### Service Status
```
× stirling-pdf.service
     Loaded: loaded (/etc/systemd/system/stirling-pdf.service; enabled; preset: ignored)
     Active: failed (Result: exit-code) since Wed 2025-08-27 04:01:28 BST; 18h ago
   Duration: 13.410s
    Process: 94943 ExecStart=/nix/store/gvljqn9lp7gz8ndgb5n3hvr3bn7jd12f-stirling-pdf-1.0.2/bin/Stirling-PDF (code=exited, status=1/FAILURE)
```

### Root Cause
**H2 Database Schema Error**: Column "U1_0.TEAM_ID" not found

```
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "U1_0.TEAM_ID" not found; SQL statement:
select u1_0.user_id,u1_0.api_key,u1_0.authenticationtype,u1_0.enabled,u1_0.is_first_login,u1_0.password,u1_0.role_name,u1_0.team_id,u1_0.username from users u1_0 where upper(u1_0.authenticationtype)=upper(?) [42122-232]
```

### Full Stack Trace
```
org.springframework.dao.DataIntegrityViolationException: could not prepare statement; SQL [select u1_0.user_id,u1_0.api_key,u1_0.authenticationtype,u1_0.enabled,u1_0.is_first_login,u1_0.password,u1_0.role_name,u1_0.team_id,u1_0.username from users u1_0 where upper(u1_0.authenticationtype)=upper(?)]; nested exception is org.hibernate.exception.GenericJDBCException: could not prepare statement
```

## Connectivity Tests

### External Access (Working)
```bash
$ curl -v https://pdf.greensroad.uk
HTTP/2 302 
location: https://greensroad.cloudflareaccess.com/cdn-cgi/access/login/...
```
✅ **External routing working** - Cloudflare Access redirects properly

### Internal Service (Failed)
```bash
$ ssh root@thor "curl -s http://127.0.0.1:8080"
Connection failed
```
❌ **Service not responding** - No local HTTP server running

### Data Directory
```bash
$ ssh root@thor "ls -la /srv/stirling-pdf/"
Data directory not found
```
❌ **Missing data directory** - May indicate initialization issues

## Analysis

This appears to be a **database migration/schema issue** where:

1. **Database Schema Mismatch**: The application expects a `team_id` column in the `users` table that doesn't exist
2. **Likely Cause**: Version upgrade or incomplete database migration
3. **Impact**: Complete service failure - application won't start

## Possible Solutions

### Option 1: Database Reset (Recommended)
- Clear existing H2 database files
- Allow Stirling PDF to recreate schema from scratch
- **Risk**: Loss of existing user accounts/settings

### Option 2: Manual Migration  
- Create missing `team_id` column manually
- **Risk**: May be other missing columns/schema changes

### Option 3: Version Rollback
- Revert to previous Stirling PDF version
- **Risk**: May lose new features/security fixes

## Recommended Fix Steps

1. **Stop service**: `systemctl stop stirling-pdf.service`
2. **Backup existing data**: `cp -r /srv/stirling-pdf /srv/stirling-pdf.backup`
3. **Clear database**: `rm -rf /srv/stirling-pdf/database/`
4. **Ensure directory permissions**: `chown -R stirling-pdf:stirling-pdf /srv/stirling-pdf`
5. **Restart service**: `systemctl start stirling-pdf.service`
6. **Verify**: `curl http://127.0.0.1:8080`

## Service Configuration Location
- **Service Module**: `modules/nixos/services/utilities/stirling-pdf.nix`
- **Configuration**: Thor host configuration
- **Port**: 8080 (internal), HTTPS via nginx proxy

---

**Priority**: Medium (service completely non-functional)
**Impact**: Users cannot access PDF processing functionality
**Complexity**: Low-Medium (database reset should resolve)

--- a/modules/nixos/services/utilities/stirling-pdf.nix
+++ b/modules/nixos/services/utilities/stirling-pdf.nix
@@ -1,3 +1,16 @@
+# Stirling PDF Service Configuration
+#
+# Database: Uses embedded H2 database stored in /var/lib/private/stirling-pdf/configs/
+# Persistence: Data is persisted via /var/lib/private mount in ZFS persist volume
+#
+# Database Reset Procedure (if schema migration fails):
+#   1. Stop service: systemctl stop stirling-pdf.service
+#   2. Backup data: tar czf /tmp/stirling-pdf-backup-$(date +%Y%m%d).tar.gz -C /var/lib/private/stirling-pdf/configs .
+#   3. Remove databases: rm -f /var/lib/private/stirling-pdf/configs/*.db /var/lib/private/stirling-pdf/configs/*.trace.db
+#   4. Remove old backups (if schema mismatch): mv /var/lib/private/stirling-pdf/configs/backup /var/lib/private/stirling-pdf/configs/backup.old
+#   5. Restart service: systemctl start stirling-pdf.service
+#
+# Note: Stirling PDF will automatically create fresh database with correct schema on startup
 {
   config,
   lib,

